samples/net/http_server/src/main.c

```diff
@@ -167,6 +176,65 @@ static int http_response_soft_404(struct http_server_ctx *ctx)
 			     HTML_FOOTER);
 }
 
+static int http_response_auth(struct http_server_ctx *ctx)
+{
+	return http_response(ctx, HTTP_STATUS_200_OK, HTML_HEADER
+			     "<h2><center>user: "HTTP_AUTH_USERNAME
+			     ", password: "HTTP_AUTH_PASSWORD
+			     "</center></h2>" HTML_FOOTER);
+}
+
+int http_response_401(struct http_server_ctx *ctx, s32_t timeout)
+{
+	/* Wait 2 secs for the reply with proper credentials */
+	return http_response_wait(ctx, HTTP_401_STATUS_US, NULL, timeout);
+}
+
+static int http_basic_auth(struct http_server_ctx *ctx)
+{
+	const char auth_str[] = "Authorization: Basic ";
+	char *ptr;
+
+	ptr = strstr(ctx->req.field_values[0].key, auth_str);
```

Comment:
Does `ctx->req.fields_values[0]` store the `\r\n` values for all the headers? If so, it might be a good idea to define `auth_str[]` as `\r\nAuthorization...`; that way it won't match headers such as `FooAuthorization: Basic ...`. Since all HTTP requests will have a `\r\n` in their headers, this will work regardless if this is the first or the last header. It might be a good idea to check for `\r\n` after the value as well, just for sanity sake.